src/output/gtk-shell.cpp

```diff
@@ -239,3 +245,31 @@ std::string wf_gtk_shell_get_custom_app_id(wf_gtk_shell *shell, wl_resource *sur
 {
     return shell->surface_app_id[surface];
 }
+
+/**
```

Comment:
this one should be in the API header, not in the source.

Reply:
Also, this should explain that such an app-id is not available for all clients and that it returns "" in these cases.